hist/hist/src/TH1.cxx

```diff
@@ -5808,6 +5808,37 @@ Bool_t TH1::Multiply(const TH1 *h1, const TH1 *h2, Double_t c1, Double_t c2, Opt
    return kTRUE;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Normalize a 1D histogram to unit integral.
+///
+/// If option contains "width", this histogram is normalized by 1/(integral * bin width),
+/// else, this histogram is normalized by 1/integral.
+
+void TH1::Normalize(Option_t *option)
+{ 
+  TString opt = option; opt.ToLower();
+  
+  if(opt.Contains("width"))
+  {
+    opt = "width"; 
+  }
+  else
+  {
+    opt = "";
+  }
+  
+  Double_t integ = this->Integral();
```

Comment:
I will pass the option "width" to the Integral function and call the virtual function 
Integral(width) removing this->

By doing this the Normalise function will work automatically also to the TH2 and TH3 cases, since they re-implement the TH1::Integral(option) function

Reply:
Then also please update the comment and indicate that it will work also for 2D and #D histograms